pkg/mtd/types.go

```diff
@@ -0,0 +1,50 @@
+// Copyright 2019 the u-root Authors. All rights reserved
+// Use of this source code is governed by a BSD-style
+// license that can be found in the LICENSE file.
+
+package mtd
+
+// Info provides information about an active MTD.
+type Info interface {
+	DevName() string
+}
+
+// Flasher defines the interface to flash drivers.
+//
+// Many devices must have lazy writes; SyncWrite should always be called
+// after a sequence of QueueWrite commands.  Close should return an
+// error if there are queued write commands. To erase a device, one
+// calls chip Blank(), QueueWrite(), and SyncWrite(). The operators are deined
+// for the Flasher, not the Chipper, since flashing can involve
+// driver-level operations such as unlocking protection bits on
+// a bridge that are more than just a chip operation.
+type Flasher interface {
+	ReadAt([]byte, int64) (int, error)
+	QueueWrite([]byte, int64) (int, error)
+	SyncWrite() error
+	Close() error
+}
+
+// Chipper is an interface to an individual chip.
+//
+// Blank will return a []byte full of "zero" values optimized for a
+// given chip -- usually, this is 0xff. Unlock and Lock are chip-level
+// operations that may be outside the scope of the driver.
+// Blocks returns a []int describing the blocks. Each int describes the
+// block size (and hence its alignment).
+// Data to be written is provided in a set of segments, but flash is
+// erased and written in blocks. Writing is not zero cost: each erase/write
+// cycle reduces chip lifetime. Data in the chip need not be erased to be
+// written: 0xee can be changed to 0xcc without an erase cycle in many parts.
+// Strategy functions will make
+// a guess at an optimal erase/write pattern based on the size of the
+// regions to be written, the content of regions, and the size of the blocks available.
+// Getting this calculation right has proven to be tricky, as it has to
+// balance time costs of writing, expected costs of too many erase cycles,
+// and several other factors I can not recall just now. Watch this space.
+type Chipper interface {
```

Comment:
s/Chipper/Chip?

Reply:
Chipper is gone. Too bad. I liked the name.